cmake/configure/configure_boost.cmake

```diff
@@ -134,42 +132,50 @@ MACRO(FEATURE_BOOST_FIND_EXTERNAL var)
         SET(${var} FALSE)
       ENDIF()
       RESET_CMAKE_REQUIRED()
-    ENDIF()
+    ENDIF() # DEAL_II_WITH_ZLIB
 
-    IF(${var} AND DEAL_II_ALLOW_PLATFORM_INTROSPECTION)
-      # We want to pass the libraries to the CMake project that tests for the
-      # BOOST Serialization bug. If we don't change the separator, only the
-      # first one is passed. This change in the separator is reverted in
-      # TestBoostBug/CMakeLists.txt
-      STRING (REPLACE ";" "|" BOOST_LIBRARIES_SEPARATED "${BOOST_LIBRARIES}")
-
-      FILE(REMOVE_RECURSE ${CMAKE_CURRENT_BINARY_DIR}/cmake/configure/TestBoostBug)
-      FILE(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/cmake/configure/TestBoostBug)
-      EXECUTE_PROCESS(
-        COMMAND ${CMAKE_COMMAND}
-                  -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
-                  -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
-                  -DBOOST_INCLUDE_DIRS=${BOOST_INCLUDE_DIRS}
-                  -DBOOST_LIBRARIES=${BOOST_LIBRARIES_SEPARATED}
-                  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/configure/TestBoostBug
-                  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/cmake/configure/TestBoostBug
-        RESULT_VARIABLE _boost_serialization_usuable
-        OUTPUT_QUIET
-        ERROR_QUIET
-      )
-      IF(${_boost_serialization_usuable} EQUAL 0)
-        EXECUTE_PROCESS(
-          COMMAND ${CMAKE_COMMAND} --build . --target run
-          WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/cmake/configure/TestBoostBug
-          RESULT_VARIABLE _boost_serialization_usuable
-          OUTPUT_QUIET
-          ERROR_QUIET
-        )
+    IF(DEAL_II_ALLOW_PLATFORM_INTROSPECTION)
+      #
+      # Test that Boost.Serialization is usable.
+      #
+      IF(NOT DEFINED BOOST_SERIALIZATION_USABLE OR NOT ${BOOST_SERIALIZATION_USABLE})
+        # Only run this check if it hasn't successfully run previously.
+        MESSAGE(STATUS "Performing Test BOOST_SERIALIZATION_USABLE")
+        IF(DEAL_II_ALLOW_PLATFORM_INTROSPECTION)
+          FILE(REMOVE_RECURSE ${CMAKE_CURRENT_BINARY_DIR}/cmake/configure/TestBoostBug)
+          FILE(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/cmake/configure/TestBoostBug)
+          EXECUTE_PROCESS(
+            COMMAND ${CMAKE_COMMAND}
+                      -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                      -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+                      "-DBOOST_INCLUDE_DIRS=${BOOST_INCLUDE_DIRS}"
+                      "-DBOOST_LIBRARIES=${BOOST_LIBRARIES}"
```

Comment:
@masterleinad The problem why you needed the workaround for the spearators were these missing quotes...